dependencies/FindLibUV.cmake

```diff
@@ -50,11 +50,13 @@ They may be set by end users to point at libuv components.
 #-----------------------------------------------------------------------------
 find_library(LibUV_LIBRARY
   NAMES uv libuv
+  PATHS $ENV{LIBUV_ROOT}/lib
```

Comment:
Mmm... this should not be needed. What error do you actually get?

Reply:
and what's your build environment?

Reply:
Ciao @ktf ,
I build on Mac OS X 10.15.5. I do not have compilation issues on O2, but on packages that depend on O2, in particular alo-o2.
So, when I build with:
`aliBuild -z --defaults o2 build alo-o2`
I get:
`-- Could NOT find LibUV (missing: LibUV_LIBRARY LibUV_INCLUDE_DIR) (found version "")`
and then:
`-- The following REQUIRED packages have not been found:`
`    * LibUV`
Now, if the library is correctly exported, I would not have to care about libUV: I only need to care about the direct dependence, which is O2.

I checked that the issue can be solved with the attached path, and then by correctly setting the LIBUV_ROOT in the module file and correctly using it in the O2 cmake:

```
diff --git a/libuv.sh b/libuv.sh
index e87bae5..5126a29 100644
--- a/libuv.sh
+++ b/libuv.sh
@@ -33,5 +33,6 @@ module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@
 module load BASE/1.0 ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION}
 # Our environment
 set LIBUV_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
+setenv LIBUV_ROOT \$LIBUV_ROOT
 prepend-path LD_LIBRARY_PATH \$LIBUV_ROOT/lib
 EoF
diff --git a/o2.sh b/o2.sh
index 2dfaa82..a71ef1a 100644
--- a/o2.sh
+++ b/o2.sh
@@ -168,8 +168,7 @@ cmake $SOURCEDIR -DCMAKE_INSTALL_PREFIX=$INSTALLROOT
       ${ALIBUILD_O2_FORCE_GPU:+-DENABLE_CUDA=ON -DENABLE_HIP=ON -DENABLE_OPENCL1=ON -DENABLE_OPENCL2=ON}  \
       ${ALIBUILD_O2_FORCE_GPU:+-DOCL2_GPUTARGET=gfx906 -DHIP_AMDGPUTARGET=gfx906 -DCUDA_COMPUTETARGET=75} \
       ${CURL_ROOT:+-DCURL_ROOT=$CURL_ROOT}                                                                \
-      ${LIBUV_ROOT:+-DLibUV_INCLUDE_DIR=$LIBUV_ROOT/include}                                              \
-      ${LIBUV_ROOT:+-DLibUV_LIBRARY=$LIBUV_ROOT/lib/libuv.$SONAME}                                        \
+      ${LIBUV_ROOT:+-DLibUV_ROOT=$LIBUV_ROOT} \
       ${ENABLE_UPGRADES:+-DENABLE_UPGRADES=ON}                                                            \
       ${ARROW_ROOT:+-Dgandiva_DIR=$ARROW_ROOT/lib/cmake/arrow}

```

Notice that the issue appears only if one uses the libUV shipped with aliBuild.
@aphecetche tested that there is no issue if libUV comes from the system (but confirms that he has the same problem as mine when he removes the libUV from the system)

Reply:
Actually adding the env variable to the `PATHS` of `find_library` should not be necessary if the env variable was named from the package name (with correct case, i.e. LibUV_ROOT and not LIBUV_ROOT) or if the corresponding CMake variable is set using `-D`, as CMake is looking to the directory indicated by the CMake variable <PackageName_ROOT> or the corresponding environment variable (https://cmake.org/cmake/help/v3.17/command/find_library.html).

So, @dstocco,  I would indeed expect the thing to work with just your `o2.sh` recipe change (and module file change). The `FindLibUV.cmake` could stay as is.

Will try to test it on my side but it's a bit cumbersome as libuv is a dependency of my editor ;-)

Reply:
Hi @aphecetche ,
indeed you're right: I tested that it is not necessary to explicitly add the LIBUV_ROOT in PATHS, since it is used by default by cmake as it corresponds to the package name.
Ok, so I close this PR and open the ones in alidist.